woocommerce/woocommerce-gutenberg-products-block
# [Store Customization MVP] Add image alts to the ai prompt if available

## What

Uses the `alt` available in some of the Verticals API images to build the prompt so the content makes sense with the image.

Also includes a pattern update which was not loading the right content.

Fixes #11084 

## Why

Since content is AI-generated, we need to avoid some images not matching the image displayed as much as possible.

## Testing Instructions


_Please consider any edge cases this change may have, and also other areas of the product this may impact._

1. Create a new [JN install](https://jurassic.ninja/) with both Jetpack and Woocommerce installed and activated.
2. Install and activate WooCommerce Blocks from this zip file: [woocommerce-gutenberg-products-block.zip](https://wcblocks.wpcomstaging.com/wp-content/uploads/woocommerce-gutenberg-products-block-11101.zip).
3. In your wp-admin, set up the JetPack connection, make sure it's successful.
4. From your admin screen, click on Tools > Plugin editor:

    Select the WooCommerce Blocks plugin and add the following within your woocommerce-blocks/woocommerce-gutenberg-products-block.php file and save:

```php
function connect_to_endpoint() {
	update_option( 'woocommerce_blocks_allow_ai_connection', true );

	$request = new WP_REST_Request( 'POST', '/wc/store/patterns' );

	$request->set_param( 'business_description', 'A store that sells organic food' );

	$response = rest_do_request( $request )->get_data();

	error_log( print_r( $response, true ) );
}

add_action( 'admin_init', 'connect_to_endpoint' );
```

5. Create a new post.
6. Insert the Featured Category Triple pattern (or any other you want with at least one image).
7. Make sure the images and content on the pattern match the business description you provided: 'A store that sells organic food'.
8. Go back to the plugin editor and update the business description over there to something else (e.g. books, clothes, sports gear etc.) and save.
9. Return to your post editor, save it, and refresh the page.
10. Test other patterns and make sure the images and their associated content makes sense.



## Screenshots or screencast



| Before | After |
| ------ | ----- |
|        |       |

## WooCommerce Visibility


Required:


## Checklist

Required:

Conditional:

## Changelog


> Add suggested changelog entry here.
